File: JINSEOP/프로그래머스_WEEK4/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    sort(phone_book.begin(), phone_book.end());
    
    for (int i = 1; i < phone_book.size(); i++) {
        if (phone_book[i - 1][0] != phone_book[i][0]) continue;
        if (phone_book[i].find(phone_book[i - 1]) != string::npos) {
            answer = false; 
            break;
        }
    }

    return answer;
}

File: DONGUK/프로그래머스_WEEK 4/전화번호 목록.py
def solution(phoneBook):
    d = {}
    phoneBook.sort(key=lambda x:len(x))
    for num in phoneBook:
        if len(d) == 0:
            d[num] = 1
        
        else:
            tmp = ''
            for i in range(len(num)):
                tmp += num[i]
                if tmp in d:
                    return False
            d[num] = 1
            
    return True